File: tests/testdata/codegen/aws-webserver-pp/go/aws-webserver.go
package main

import (
	"github.com/khulnasoft/khulnasoft-aws/sdk/v5/go/aws"
	"github.com/khulnasoft/khulnasoft-aws/sdk/v5/go/aws/ec2"
	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		// Create a new security group for port 80.
		securityGroup, err := ec2.NewSecurityGroup(ctx, "securityGroup", &ec2.SecurityGroupArgs{
			Ingress: ec2.SecurityGroupIngressArray{
				&ec2.SecurityGroupIngressArgs{
					Protocol: khulnasoft.String("tcp"),
					FromPort: khulnasoft.Int(0),
					ToPort:   khulnasoft.Int(0),
					CidrBlocks: khulnasoft.StringArray{
						khulnasoft.String("0.0.0.0/0"),
					},
				},
			},
		})
		if err != nil {
			return err
		}
		// Get the ID for the latest Amazon Linux AMI.
		ami, err := aws.GetAmi(ctx, &aws.GetAmiArgs{
			Filters: []aws.GetAmiFilter{
				{
					Name: "name",
					Values: []string{
						"amzn-ami-hvm-*-x86_64-ebs",
					},
				},
			},
			Owners: []string{
				"137112412989",
			},
			MostRecent: khulnasoft.BoolRef(true),
		}, nil)
		if err != nil {
			return err
		}
		// Create a simple web server using the startup script for the instance.
		server, err := ec2.NewInstance(ctx, "server", &ec2.InstanceArgs{
			Tags: khulnasoft.StringMap{
				"Name": khulnasoft.String("web-server-www"),
			},
			InstanceType: khulnasoft.String(ec2.InstanceType_T2_Micro),
			SecurityGroups: khulnasoft.StringArray{
				securityGroup.Name,
			},
			Ami:      khulnasoft.String(ami.Id),
			UserData: khulnasoft.String("#!/bin/bash\necho \"Hello, World!\" > index.html\nnohup python -m SimpleHTTPServer 80 &\n"),
		})
		if err != nil {
			return err
		}
		ctx.Export("publicIp", server.PublicIp)
		ctx.Export("publicHostName", server.PublicDns)
		return nil
	})
}

File: tests/testdata/codegen/aws-webserver-pp/python/aws-webserver.py
import khulnasoft
import khulnasoft_aws as aws

# Create a new security group for port 80.
security_group = aws.ec2.SecurityGroup("securityGroup", ingress=[{
    "protocol": "tcp",
    "from_port": 0,
    "to_port": 0,
    "cidr_blocks": ["0.0.0.0/0"],
}])
# Get the ID for the latest Amazon Linux AMI.
ami = aws.get_ami(filters=[{
        "name": "name",
        "values": ["amzn-ami-hvm-*-x86_64-ebs"],
    }],
    owners=["137112412989"],
    most_recent=True)
# Create a simple web server using the startup script for the instance.
server = aws.ec2.Instance("server",
    tags={
        "Name": "web-server-www",
    },
    instance_type=aws.ec2.InstanceType.T2_MICRO,
    security_groups=[security_group.name],
    ami=ami.id,
    user_data="""#!/bin/bash
echo "Hello, World!" > index.html
nohup python -m SimpleHTTPServer 80 &
""")
khulnasoft.export("publicIp", server.public_ip)
khulnasoft.export("publicHostName", server.public_dns)
